code-challenges.test.js

```diff
@@ -21,30 +20,120 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+    // PSUEDO CODE
+    // create function that takes an array
+    // removes first item from array
+    // shuffles array
+    // return new array
+    // for the test, expect statement should be flexible on order of returened array in expected output
+
+    describe('rm_First_Element_Shuffle_Array', () => {
+        it('takes an array, removes the first element and shuffles the remaining contents', () => {
+            expect(rm_First_Element_Shuffle_Array(colors1)).not.toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]));
+            expect(rm_First_Element_Shuffle_Array(colors2)).not.toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]));
+        })
+    })
+    
+    //  GOOD FAIL:
+    //  rm_first_element_shuffle_array › takes an array, removes the first element and shuffles the remaining contents
+    //  ReferenceError: rm_first_element_shuffle_array is not defined
+    
 
 // b) Create the function that makes the test pass.
 
+const rm_First_Element_Shuffle_Array = (array) => {
+    let rm_First_Element = array.pop()
+    let shuffle = (arr) => {
+        let currentIndex = arr.length, randomIndex;
+        // While there remain elements to shuffle.
+        while (currentIndex !=0) {
+
+            // Pick a remaining element.
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex--;
+
+            // And swap it with the current element.
+            [arr[currentIndex], arr[randomIndex]] = [arr[randomIndex], arr[currentIndex]];
+        }
+        return arr
+    }
+    shuffle(rm_First_Element)
+}
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the net total of votes.
 
+// PSEUDO CODE
+// create a function that takes an object of key/value vote: up or down
+// return total votes
```

Comment:
suggest including how you will abstract the values and the type of calculation you will use to obtain the total